yt_dlp/extractor/srmediathek.py

```diff
@@ -1,57 +1,100 @@
-        'params': {
-            # m3u8 download
-            'skip_download': True,
+            'id': '153853',
+            'ext': 'mp3',
+            'title': 'Kappes, Klöße, Kokosmilch: Bruschetta mit Nduja',
+            'channel': 'SR 3',
+            'description': 'md5:3935798de3562b10c4070b408a15e225',
+            'duration': 139.0,
+            'release_date': '20250523',
+            'series': 'Kappes, Klöße, Kokosmilch',
+            'series_id': 'SR3_KKK_A',
+            'thumbnail': r're:https?://.+\.jpg',
         },
     }, {
-        'url': 'http://sr-mediathek.de/index.php?seite=7&id=7480',
-        'only_matching': True,
+        'url': 'https://www.sr-mediathek.de/index.php?seite=7&id=31406&pnr=&tbl=pf',
+        'info_dict': {
+            'id': '31406',
+            'ext': 'mp3',
+            'title': 'Das Leben schwer nehmen, ist einfach zu anstrengend',
+            'channel': 'SR 1',
+            'description': 'md5:3e03fd556af831ad984d0add7175fb0c',
+            'duration': 1769.0,
+            'release_date': '20230717',
+            'series': 'Abendrot',
+            'series_id': 'SR1_AB_P',
+            'thumbnail': r're:https?://.+\.jpg',
+        },
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
+        description = self._og_search_description(webpage)
 
-        if '>Der gew&uuml;nschte Beitrag ist leider nicht mehr verf&uuml;gbar.<' in webpage:
+        if description == 'Der gewünschte Beitrag ist leider nicht mehr vorhanden.':
             raise ExtractorError(f'Video {video_id} is no longer available', expected=True)
 
-        media_collection_url = self._search_regex(
-            r'data-mediacollection-ardplayer="([^"]+)"', webpage, 'media collection url')
-        info = self._extract_media_info(media_collection_url, webpage, video_id)
-        info.update({
+        player_url = traverse_obj(webpage, (
+            {find_element(tag='div', id=f'player{video_id}', html=True)},
+            {extract_attributes}, 'data-mediacollection-ardplayer',
+            {lambda x: self._proto_relative_url(x)},
+        ))
+        article = traverse_obj(webpage, (
+            {find_element(cls='article__content')},
+            {find_element(tag='p')}, {clean_html},
+        ))
+        cls_common = 'teaser__image__caption__text teaser__image__caption__text--'
+
+        return {
+            **self._extract_media_info(player_url, webpage, video_id),
             'id': video_id,
-            'title': get_element_by_attribute('class', 'ardplayer-title', webpage),
-            'description': self._og_search_description(webpage),
+            'title': traverse_obj(webpage, (
+                {find_element(cls='ardplayer-title')}, {str.strip})),
+            'channel': traverse_obj(webpage, (
+                {find_element(cls=f'{cls_common}subheadline')},
+                {lambda x: x.split('|')[0]}, {str.strip})),
```

Comment:
I'd just use `clean_html` in these traversals to be safe (it will take care of the whitespace too).

Instead of `lambda x: x.split('|')[0]` could we use `trim_str` here (e.g. if there's a fixed string after the `|`)?
```suggestion
            'title': traverse_obj(webpage, (
                {find_element(cls='ardplayer-title')}, {clean_html})),
            'channel': traverse_obj(webpage, (
                {find_element(cls=f'{cls_common}subheadline')},
                {lambda x: x.split('|')[0]}, {clean_html})),
```

Reply:
> Instead of `lambda x: x.split('|')[0]` could we use `trim_str` here (e.g. if there's a fixed string after the `|`)?

substring after `|` isn’t fixed in length; it’s variable